tests/benchmark/spaces_benchmarks/bm_spaces.h

```diff
@@ -41,47 +41,54 @@
         }                                                                                          \
     }
 
-// dim_opt:   number of elements in 512 bits.
-//            i.e. 512/sizeof(type): FP32 = 16, FP64 = 8, BF16 = 32 ...
-//            The is the number of elements calculated in each distance function loop,
-//            regardless of the arch optimization type.
-// Run each function for {1, 4, 16} iterations.
-#define EXACT_512BIT_PARAMS(dim_opt) RangeMultiplier(4)->Range(dim_opt, 400)
-
-// elem_per_128_bits:   dim_opt / 4.  FP32 = 4, FP64 = 2, BF16 = 8...
-// Dimensions to test 128 bit chunks.
-// Run each function at least one full 512 bits iteration + 1/2/3 iterations of 128 bit chunks.
-#define EXACT_128BIT_PARAMS(elem_per_128_bits)                                                     \
-    DenseRange(128 + elem_per_128_bits, 128 + 3 * elem_per_128_bits, elem_per_128_bits)
-
-// Run each function at least one full 512 bits iteration + (1 * elements : elem_per_128_bits *
-// elements) FP32 = residual = 1,2,3, FP64 = residual = 1, BF16 = residual = 1,2,3,4,5,6,7...
-#define RESIDUAL_PARAMS(elem_per_128_bits) DenseRange(128 + 1, 128 + elem_per_128_bits - 1, 1)
-
 #define INITIALIZE_BM(bm_class, type_prefix, arch, metric, bm_name, arch_supported)                \
     BENCHMARK_DISTANCE_F(bm_class, type_prefix, arch, metric, bm_name, arch_supported)             \
     BENCHMARK_REGISTER_F(bm_class, type_prefix##_##arch##_##metric##_##bm_name)                    \
         ->ArgName("Dimension")                                                                     \
         ->Unit(benchmark::kNanosecond)
 
+/**
+ * @param dim_opt: smallest dimension with no residual.
+ * i.e smallest dimension to satisfy:
+ * dim % num_elements_in_512_bits == 0.
```

Comment:
Overall, this is a good comment, but this line confused me at first. Consider using bullet points/numbered list to imply there is more than one condition it has to satisfy